<!DOCTYPE html>
<html lang="en">

  <head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta name="description" content="">
    <meta name="author" content="">

    <title>eQTLGen - Publications</title>

    <!-- Bootstrap core CSS -->
    <link href="css/vendor/bootstrap/bootstrap.min.css" rel="stylesheet">

    <!-- Custom styles for this template -->
    <link href="css/eqtlgen.css" rel="stylesheet">

  </head>

  <body>

    <!-- Navigation -->
    <nav class="navbar fixed-top navbar-expand-lg navbar-dark bg-dark fixed-top">
      <div class="container">
        <a class="navbar-brand" href="phase1.html">eQTLGen phase I</a>
        <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarResponsive" aria-controls="navbarResponsive" aria-expanded="false" aria-label="Toggle navigation">
          <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="navbarResponsive">
          <ul class="navbar-nav ml-auto">
            <li class="nav-item">
              <a class="nav-link" href="cis-eqtls.html"><em>Cis</em>-eQTLs</a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="trans-eqtls.html"><em>Trans</em>-eQTLs</a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="eqts.html">eQTS</a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="replications.html">Replications</a>
            </li>
            <li class="nav-item">
              <a class="nav-link active" href="publications.html">Publications</a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="sc/index.html">sc-eQTLGen</a>
            </li>
          </ul>
        </div>
      </div>
    </nav>

    <!-- Page Content -->
    <div class="container" style="margin-top: 75px">

      <h2 class="my-4">Publications</h2>

      An overview of papers, tools and databases using or citing eQTLGen Consortium data.<br/>
      Last update: 21 April 2020.<br/><br/>

      <h5>Citation</h5>
      If you use the data on this website, please cite our paper:<br/>
      Võsa, U., Claringbould, A., (…), Franke, L.; 2018; Unraveling the polygenic architecture of complex traits using blood eQTL meta-analysis<br/><br/>

      <!-- regex doc to html: ^(.+)(http.+) -> <li>$1<a target="_blank" href="$2">$2</a></li> -->
      <h5>Papers using eQTLGen data (before publication)</h5>
      <ol>
        <li>Karlsson Linnér, R., et al. (2019). Genome-wide association analyses of risk tolerance and risky behaviors in over 1 million individuals identify hundreds of loci and shared genetic influences. Nature Genetics, 51(2), 245–257. <a target="_blank" href="http://doi.org/10.1038/s41588-018-0309-3">http://doi.org/10.1038/s41588-018-0309-3</a></li>
        <li>Lepik, K., et al. (2017). C-reactive protein upregulates the whole blood expression of CD59 - an integrative analysis. PLOS Computational Biology, 13(9), e1005766. <a target="_blank" href="http://doi.org/10.1371/journal.pcbi.1005766">http://doi.org/10.1371/journal.pcbi.1005766</a></li>
        <li>Männik, K., et al. (2019). Leveraging biobank-scale rare and common variant analyses to identify ASPHD1 as the main driver of reproductive traits in the 16p11.2 locus. BioRxiv, 716415. <a target="_blank" href="https://doi.org/10.1101/716415">https://doi.org/10.1101/716415</a></li>
        <li>Porcu, E., et al. (2019). Mendelian randomization integrating GWAS and eQTL data reveals genetic determinants of complex and clinical traits. Nature Communications, 10(1), 3300. <a target="_blank" href="https://doi.org/10.1038/s41467-019-10936-0">https://doi.org/10.1038/s41467-019-10936-0</a></li>
        <li>Thompson, D., et al. (2019). Genetic predisposition to mosaic Y chromosome loss in blood is associated with genomic instability in other tissues and susceptibility to non-haematological cancers. BioRxiv, 514026. <a target="_blank" href="http://doi.org/10.1101/514026">http://doi.org/10.1101/514026</a></li>
        <li>Timmers, P. R., et al. (2019). Genomics of 1 million parent lifespans implicates novel pathways and common diseases and distinguishes survival chances. ELife, 8. <a target="_blank" href="http://doi.org/10.7554/eLife.39856">http://doi.org/10.7554/eLife.39856</a></li>
        <li>Qi, T., et al. (2018). Identifying gene targets for brain-related traits using transcriptomic and methylomic data from blood. Nature Communications, 9(1), 2282. <a target="_blank" href="http://doi.org/10.1038/s41467-018-04558-1">http://doi.org/10.1038/s41467-018-04558-1</a></li>
        <li>Xue, A., et al. (2018). Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes. Nature Communications, 9(1), 2941. <a target="_blank" href="http://doi.org/10.1038/s41467-018-04951-w">http://doi.org/10.1038/s41467-018-04951-w</a></li>
        <li>Wray, N. R., et al. (2018). Genome-wide association analyses identify 44 risk variants and refine the genetic architecture of major depression. Nature Genetics, 50(5), 668–681. <a target="_blank" href="http://doi.org/10.1038/s41588-018-0090-3">http://doi.org/10.1038/s41588-018-0090-3</a></li>
      </ol>

      <h5>Papers using eQTLGen data (after publication)</h5>
      <ol>
        <li value="10">Akosile, W., et al. (2019). NLRP3 is associated with coronary artery disease in Vietnam veterans. Gene, 144163. <a target="_blank" href="https://doi.org/10.1016/j.gene.2019.144163">https://doi.org/10.1016/j.gene.2019.144163</a></li>
        <li>Allsup, D., et al. (2019). Multicentre Genome Wide Association Study Identifies Risk Alleles for Progressive Chronic Lymphocytic Leukaemia. Blood, 134(Supplement_1), 1740–1740. <a target="_blank" href="https://doi.org/10.1182/blood-2019-122037">https://doi.org/10.1182/blood-2019-122037</a></li>
        <li>Andrews, S. J., & Goate, A. (2019). Mendelian randomization indicates that TNF is not causally associated with Alzheimer’s disease. Neurobiology of Aging. <a target="_blank" href="https://doi.org/10.1016/j.neurobiolaging.2019.09.003">https://doi.org/10.1016/j.neurobiolaging.2019.09.003</a></li>
        <li>Arbeeva, L., et al. (2020). Genome-wide meta-analysis identified novel variant associated with hallux valgus in Caucasians. Journal of Foot and Ankle Research, 13(1), 11. <a target="_blank" href="https://doi.org/10.1186/s13047-020-0379-1">https://doi.org/10.1186/s13047-020-0379-1</a></li>
        <li>Arvanitis, M., et al. (2020). Genome-wide association and multi-omic analyses reveal ACTN2 as a gene linked to heart failure. Nature Communications, 11(1), 1122. <a target="_blank" href="https://doi.org/10.1038/s41467-020-14843-7">https://doi.org/10.1038/s41467-020-14843-7</a></li>
        <li>Cai, M., et al. (2020). IGREX for quantifying the impact of genetically regulated expression on phenotypes. NAR Genomics and Bioinformatics, 2(1). <a target="_blank" href="https://doi.org/10.1093/NARGAB/LQAA010">https://doi.org/10.1093/NARGAB/LQAA010</a></li>
        <li>Cai, X.-Y., et al. (2019). GWAS Follow-up Study Discovers a Novel Genetic Signal on 10q21.2 for Atopic Dermatitis in Chinese Han Population. Frontiers in Genetics, 10, 174. <a target="_blank" href="http://doi.org/10.3389/fgene.2019.00174">http://doi.org/10.3389/fgene.2019.00174</a></li>
        <li>Carmeli, C., et al (2020). Gene regulation contributes to explain the impact of early life socioeconomic disadvantage on adult inflammatory levels in two European cohort studies. MedRxiv, 2020.04.03.20050872. <a target="_blank" href="https://doi.org/10.1101/2020.04.03.20050872">https://doi.org/10.1101/2020.04.03.20050872</a></li>
        <li>Censin, J. C., et al. (2020). Colocalization highlights genes in hypothalamic–pituitary–gonadal axis as potentially mediating polycystic ovary syndrome risk. BioRxiv, 2020.01.10.901116. <a target="_blank" href="https://doi.org/10.1101/2020.01.10.901116">https://doi.org/10.1101/2020.01.10.901116</a></li>
        <li>Chauquet, S., et al. (2020). Investigating the potential effect of antihypertensive medication on psychiatric disorders: a mendelian randomisation study. MedRxiv, 2020.03.19.20039412. <a target="_blank" href="https://doi.org/10.1101/2020.03.19.20039412">https://doi.org/10.1101/2020.03.19.20039412</a></li>
        <li>Fernandez-Jimenez, N., & Bilbao, J. R. (2019). Mendelian Randomization analysis of celiac GWAS reveals a blood expression signature with diagnostic potential in absence of gluten consumption. Human Molecular Genetics. <a target="_blank" href="https://doi.org/10.1093/hmg/ddz113">https://doi.org/10.1093/hmg/ddz113</a></li>
        <li>Ferreira, R. C., et al. (2019). Chronic Immune Activation in Systemic Lupus Erythematosus and the Autoimmune PTPN22 Trp620 Risk Allele Drive the Expansion of FOXP3+ Regulatory T Cells and PD-1 Expression. Frontiers in Immunology, 10. <a target="_blank" href="https://doi.org/10.3389/fimmu.2019.02606">https://doi.org/10.3389/fimmu.2019.02606</a></li>
        <li>Fine, R. S., et al. (2019). Benchmarker: An Unbiased, Association-Data-Driven Strategy to Evaluate Gene Prioritization Algorithms. The American Journal of Human Genetics, 104(6), 1025–1039. <a target="_blank" href="https://doi.org/10.1016/j.ajhg.2019.03.027">https://doi.org/10.1016/j.ajhg.2019.03.027</a></li>
        <li>Foley, C. N., et al. (2019). A fast and efficient colocalization algorithm for identifying shared genetic risk factors across multiple traits. BioRxiv, 592238. <a target="_blank" href="http://doi.org/10.1101/592238">http://doi.org/10.1101/592238</a></li>
        <li>Folkersen, L., et al. (2020). Genomic evaluation of circulating proteins for drug target characterisation and precision medicine. BioRxiv, 2020.04.03.023804. <a target="_blank" href="https://doi.org/10.1101/2020.04.03.023804">https://doi.org/10.1101/2020.04.03.023804</a></li>
        <li>Gibson, J., et al. (2019). A meta-analysis of genome-wide association studies of epigenetic age acceleration. PLOS Genetics, 15(11), e1008104. <a target="_blank" href="https://doi.org/10.1371/journal.pgen.1008104">https://doi.org/10.1371/journal.pgen.1008104</a></li>
        <li>Grenn, F. P., et al. (2020). The Parkinson’s Disease GWAS Locus Browser. BioRxiv Genetics, 2020.04.01.020404. <a target="_blank" href="https://doi.org/10.1101/2020.04.01.020404">https://doi.org/10.1101/2020.04.01.020404</a></li>
        <li>Han, Y., et al. (2020). Genome-wide analysis highlights contribution of immune system pathways to the genetic architecture of asthma. Nature Communications, 11(1), 1776. <a target="_blank" href="https://doi.org/10.1038/s41467-020-15649-3">https://doi.org/10.1038/s41467-020-15649-3</a></li>
        <li>Hillary, R. F., et al. (2019). Genome and epigenome wide studies of neurological protein biomarkers in the Lothian Birth Cohort 1936. Nature Communications, 10(1). <a target="_blank" href="https://doi.org/10.1038/s41467-019-11177-x">https://doi.org/10.1038/s41467-019-11177-x</a></li>
        <li>Hillary, R. F., et al (2020). Integrative omics approach to identify the molecular architecture of inflammatory protein levels in healthy older adults. BioRxiv, 2020.02.17.952135. <a target="_blank" href="https://doi.org/10.1101/2020.02.17.952135">https://doi.org/10.1101/2020.02.17.952135</a></li>
        <li>Hobbs, B. D., et al. (2019). Overlap of Genetic Risk Between Interstitial Lung Abnormalities and Idiopathic Pulmonary Fibrosis. American Journal of Respiratory and Critical Care Medicine. <a target="_blank" href="https://doi.org/10.1164/rccm.201903-0511oc">https://doi.org/10.1164/rccm.201903-0511oc</a></li>
        <li>Hyvärinen, K., et al. (2020). Meta-Analysis of Genome-Wide Association and Gene Expression Studies Implicates Donor T Cell Function and Cytokine Pathways in Acute GvHD. Frontiers in Immunology, 11. <a target="_blank" href="https://doi.org/10.3389/fimmu.2020.00019">https://doi.org/10.3389/fimmu.2020.00019</a></li>
        <li>Inshaw, J. R. J., et al. (2019). Genetic Variants Predisposing Most Strongly to Type 1 Diabetes Diagnosed Under Age 7 Years Lie Near Candidate Genes That Function in the Immune System and in Pancreatic β-Cells. Diabetes Care, dc190803. <a target="_blank" href="https://doi.org/10.2337/dc19-0803">https://doi.org/10.2337/dc19-0803</a></li>
        <li>Iwaki, H., et al. (2019). Genetic risk of Parkinson disease and progression: Neurology Genetics, 5(4), e348. <a target="_blank" href="https://doi.org/10.1212/NXG.0000000000000348">https://doi.org/10.1212/NXG.0000000000000348</a></li>
        <li>Iwaki, H., et al. (2019). Genomewide association study of Parkinson’s disease clinical biomarkers in 12 longitudinal patients’ cohorts. Movement Disorders, 34(12), 1839–1850. <a target="_blank" href="https://doi.org/10.1002/mds.27845">https://doi.org/10.1002/mds.27845</a></li>
        <li>Jabbari, E., et al. (2020). Common variation at the LRRK2 locus is associated with survival in the primary tauopathy progressive supranuclear palsy. BioRxiv, 2020.02.04.932335. <a target="_blank" href="https://doi.org/10.1101/2020.02.04.932335">https://doi.org/10.1101/2020.02.04.932335</a></li>
        <li>Jacobs, B. M., et al. (2020). Summary-data-based mendelian randomisation reveals druggable targets for multiple sclerosis. BioRxiv, 2020.01.20.907451. <a target="_blank" href="https://doi.org/10.1101/2020.01.20.907451">https://doi.org/10.1101/2020.01.20.907451</a></li>
        <li>Jaeger, M., et al. (2019). A genome-wide functional genomics approach identifies susceptibility pathways to fungal bloodstream infection in humans. The Journal of Infectious Diseases. <a target="_blank" href="http://doi.org/10.1093/infdis/jiz206">http://doi.org/10.1093/infdis/jiz206</a></li>
        <li>Jamieson, E., et al. (2019). Smoking, DNA methylation and lung function: a Mendelian randomization analysis to investigate causal relationships. bioRxiv. <a target="_blank" href="https://doi.org/10.1101/19003335">https://doi.org/10.1101/19003335</a></li>
        <li>Kamat, M. A., et al. (2019). PhenoScanner V2: an expanded tool for searching human genotype–phenotype associations. Bioinformatics, 35(22), 4851–4853. <a target="_blank" href="https://doi.org/10.1093/bioinformatics/btz469">https://doi.org/10.1093/bioinformatics/btz469</a></li>
        <li>Karaesmen, E., et al. (2019). Multiple functional variants in the IL1RL1 region are pretransplant markers for risk of GVHD and infection deaths. Blood Advances, 3(16), 2512–2524. <a target="_blank" href="https://doi.org/10.1182/bloodadvances.2019000075">https://doi.org/10.1182/bloodadvances.2019000075</a></li>
        <li>Ke, J., et al. (2019). Evaluation of polymorphisms in microRNA‐binding sites and pancreatic cancer risk in Chinese population. Journal of Cellular and Molecular Medicine, jcmm.14906. <a target="_blank" href="https://doi.org/10.1111/jcmm.14906">https://doi.org/10.1111/jcmm.14906</a></li>
        <li>Kolberg, L., et al. (2020). Co-expression analysis reveals interpretable gene modules controlled by trans-acting genetic variants. BioRxiv, 2020.04.22.055335. <a target="_blank" href="https://doi.org/10.1101/2020.04.22.055335">https://doi.org/10.1101/2020.04.22.055335</a></li>
        <li>Krebs, K., et al. (2020). Genome-wide study identifies association between HLA-B*55:01 and penicillin allergy. BioRxiv, 2020.02.27.967497. <a target="_blank" href="https://doi.org/10.1101/2020.02.27.967497">https://doi.org/10.1101/2020.02.27.967497</a></li>
        <li>Le, K. T. T., et al. (2019). Functional Annotation of Genetic Loci Associated With Sepsis Prioritizes Immune and Endothelial Cell Pathways. Frontiers in Immunology, 10, 1949. <a target="_blank" href="https://doi.org/10.3389/fimmu.2019.01949">https://doi.org/10.3389/fimmu.2019.01949</a></li>
        <li>Le Guen, et al. (2020). Enhancer locus in ch14q23.1 modulates brain asymmetric temporal regions involved in language processing. <a target="_blank" href="https://doi.org/10.1101/539189">https://doi.org/10.1101/539189</a></li>
        <li>Leyden, G. M., et al. (2020). A factorial Mendelian randomization study to systematically prioritize genetic targets for the treatment of cardiovascular disease. MedRxiv, 2020.02.16.20023010. <a target="_blank" href="https://doi.org/10.1101/2020.02.16.20023010">https://doi.org/10.1101/2020.02.16.20023010</a></li>
        <li>Liang, Y., et al. (2020). Scalable unified framework of total and allele-specific counts for cis-QTL, fine-mapping, and prediction. BioRxiv, 2020.04.22.050666. <a target="_blank" href="https://doi.org/10.1101/2020.04.22.050666">https://doi.org/10.1101/2020.04.22.050666</a></li>
        <li>Liu, G., et al. (2020). rs4147929 variant minor allele increases ABCA7 gene expression and ABCA7 shows increased gene expression in Alzheimer’s disease patients compared with controls. Acta Neuropathologica, 1–4. <a target="_blank" href="https://doi.org/10.1007/s00401-020-02135-9">https://doi.org/10.1007/s00401-020-02135-9</a></li>
        <li>Madan, N., et al. (2019). Functionalization of CD36 Cardiovascular Disease and Expression Associated Variants by Interdisciplinary High Throughput Analysis. PLOS Genetics, 15(7), e1008287. <a target="_blank" href="https://doi.org/10.1371/journal.pgen.1008287">https://doi.org/10.1371/journal.pgen.1008287</a></li>
        <li>Marderstein, A. R., et al. (2019). Age, Sex, and Genetics Influence the Abundance of Infiltrating Immune Cells in Human Tissues. BioRxiv, 614305. <a target="_blank" href="http://doi.org/10.1101/614305">http://doi.org/10.1101/614305</a></li>
        <li>Milet, J., et al. (2019). First genome-wide association study of non-severe malaria in two birth cohorts in Benin. Human Genetics. <a target="_blank" href="https://doi.org/10.1007/s00439-019-02079-5">https://doi.org/10.1007/s00439-019-02079-5</a></li>
        <li>Nath, A. P., et al. (2019). Multivariate Genome-wide Association Analysis of a Cytokine Network Reveals Variants with Widespread Immune, Haematological, and Cardiometabolic Pleiotropy. The American Journal of Human Genetics, 105(6), 1076–1090. <a target="_blank" href="https://doi.org/10.1016/j.ajhg.2019.10.001">https://doi.org/10.1016/j.ajhg.2019.10.001</a></li>
        <li>Nalls, M. A., et al. (2019). Identification of novel risk loci, causal insights, and heritable risk for Parkinson’s disease: a meta-analysis of genome-wide association studies. The Lancet Neurology, 18(12), 1091–1102. <a target="_blank" href="https://doi.org/10.1016/S1474-4422(19)30320-5">https://doi.org/10.1016/S1474-4422(19)30320-5</a></li>
        <li>Nudel, R., et al. (2020). A large population-based investigation into the genetics of susceptibility to gastrointestinal infections and the link between gastrointestinal infections and mental illness. Human Genetics, 1–12. <a target="_blank" href="https://doi.org/10.1007/s00439-020-02140-8">https://doi.org/10.1007/s00439-020-02140-8</a></li>
        <li>O’Mara, T. A., et al. (2019). Analysis of Promoter-Associated Chromatin Interactions Reveals Biologically Relevant Candidate Target Genes at Endometrial Cancer Risk Loci. Cancers, 11(10), 1440. <a target="_blank" href="https://doi.org/10.3390/cancers11101440">https://doi.org/10.3390/cancers11101440</a></li>
        <li>Patrick, M. T., et al. (2019). Integrative Approach to Reveal Cell Type Specificity and Gene Candidates for Psoriatic Arthritis Outside the MHC. Frontiers in Genetics, 10, 304. <a target="_blank" href="https://doi.org/10.3389/fgene.2019.00304">https://doi.org/10.3389/fgene.2019.00304</a></li>
        <li>Porcu, E., et al. (2020). The role of gene expression on human sexual dimorphism: too early to call. BioRxiv, 2020.04.15.042986. <a target="_blank" href="https://doi.org/10.1101/2020.04.15.042986">https://doi.org/10.1101/2020.04.15.042986</a></li>
        <li>Raffield, L. M., et al. (2020). Allelic Heterogeneity at the CRP Locus Identified by Whole-Genome Sequencing in Multi-ancestry Cohorts. American Journal of Human Genetics, 106(1), 112–120. <a target="_blank" href="https://doi.org/10.1016/j.ajhg.2019.12.002">https://doi.org/10.1016/j.ajhg.2019.12.002</a></li>
        <li>Revez, J. A., et al. (2020). Genome-wide association study identifies 143 loci associated with 25 hydroxyvitamin D concentration. Nature Communications, 11(1), 1–12. <a target="_blank" href="https://doi.org/10.1038/s41467-020-15421-7">https://doi.org/10.1038/s41467-020-15421-7</a></li>
        <li>Richardson, T. G., et al. (2019). A transcriptome-wide Mendelian randomization study to uncover tissue-dependent regulatory mechanisms across the human phenome. BioRxiv, 563379. <a target="_blank" href="http://doi.org/10.1101/563379">http://doi.org/10.1101/563379</a></li>
        <li>Saferali, A., et al. (2019). Analysis of genetically driven alternative splicing identifies FBXO38 as a novel COPD susceptibility gene. PLOS Genetics, 15(7), e1008229. <a target="_blank" href="https://doi.org/10.1371/journal.pgen.1008229">https://doi.org/10.1371/journal.pgen.1008229</a></li>
        <li>Shadrin, A. A., et al. (2019). A genome-wide genetic pleiotropy approach identified shared loci between multiple system atrophy and inflammatory bowel disease. BioRxiv, 751354. <a target="_blank" href="https://doi.org/10.1101/751354">https://doi.org/10.1101/751354</a></li>
        <li>Shah, S., et al. (2020). Genome-wide association and Mendelian randomisation analysis provide insights into the pathogenesis of heart failure. Nature Communications, 1–12. <a target="_blank" href="https://doi.org/10.1038/s41467-019-13690-5">https://doi.org/10.1038/s41467-019-13690-5</a></li>
        <li>Shi, C., et al. (2020). An active chromatin interactome elucidates the biological mechanisms underlying genetic risk factors of dermatological conditions in disease relevant cell lines. BioRxiv, 2020.03.05.973271. <a target="_blank" href="https://doi.org/10.1101/2020.03.05.973271">https://doi.org/10.1101/2020.03.05.973271</a></li>
        <li>Stevelink, R., et al. (2019). Assessing the genetic association between vitamin B6 metabolism and genetic generalized epilepsy. Molecular Genetics and Metabolism Reports, 21, 100518. <a target="_blank" href="https://doi.org/10.1016/J.YMGMR.2019.100518">https://doi.org/10.1016/J.YMGMR.2019.100518</a></li>
        <li>Storm, C. S. et al. (2020). Using Mendelian randomization to understand and develop treatments for neurodegenerative disease. Brain Communications. <a target="_blank" href="https://doi.org/10.1093/BRAINCOMMS/FCAA031">https://doi.org/10.1093/BRAINCOMMS/FCAA031</a></li>
        <li>Tan, J. Y., & Marques, A. C. (2020). The activity of human enhancers is modulated by the splicing of their associated lncRNAs. BioRxiv, 2020.04.17.045971. <a target="_blank" href="https://doi.org/10.1101/2020.04.17.045971">https://doi.org/10.1101/2020.04.17.045971</a></li>
        <li>Tang, H., et al. (2019). Validation of genetic associations with acute GVHD and nonrelapse mortality in DISCOVeRY-BMT. Blood Advances, 3(15), 2337–2341. <a target="_blank" href="https://doi.org/10.1182/bloodadvances.2019000052">https://doi.org/10.1182/bloodadvances.2019000052</a></li>
        <li>Taub, M. A., et al. (2019). Novel genetic determinants of telomere length from a multi-ethnic analysis of 75,000 whole genome sequences in TOPMed. BioRxiv, 749010. <a target="_blank" href="https://doi.org/10.1101/749010">https://doi.org/10.1101/749010</a></li>
        <li>Venkateswaran, S., et al. (2019). Neutrophil GM-CSF signaling in inflammatory bowel disease patients is influenced by non-coding genetic variants. Scientific Reports, 9(1), 9168. <a target="_blank" href="https://doi.org/10.1038/s41598-019-45701-2">https://doi.org/10.1038/s41598-019-45701-2</a></li>
        <li>de Vries, D. H., et al. (2020). Integrating GWAS with bulk and single-cell RNA-sequencing reveals a role for LY86 in the anti-Candida host response. PLOS Pathogens, 16(4), e1008408. <a target="_blank" href="https://doi.org/10.1371/journal.ppat.1008408">https://doi.org/10.1371/journal.ppat.1008408</a></li>
        <li>Vuckovic, et al. (2020). The Polygenic and Monogenic Basis of Blood Traits and Diseases. MedRxiv, 2020.02.02.20020065. <a target="_blank" href="https://doi.org/10.1101/2020.02.02.20020065">https://doi.org/10.1101/2020.02.02.20020065</a></li>
        <li>Wang, J., et al. (2019). Genome-wide association analyses identify variants in IRF4 associated with Acute Myeloid Leukemia and Myelodysplastic Syndrome susceptibility. BioRxiv, 773952. <a target="_blank" href="https://doi.org/10.1101/773952">https://doi.org/10.1101/773952</a></li>
        <li>Wang, X., et al., (2019). Integrating genome-wide association study and expression quantitative trait loci data identifies NEGR1 as a causal risk gene of major depression disorder. Journal of Affective Disorders. <a target="_blank" href="https://doi.org/10.1016/j.jad.2019.11.116">https://doi.org/10.1016/j.jad.2019.11.116</a></li>
        <li>Wheeler, H. E., et al. (2019). Imputed gene associations identify replicable trans-acting genes enriched in transcription pathways and complex traits. Genetic Epidemiology, 1-13. <a target="_blank" href="http://doi.org/10.1002/gepi.22205">http://doi.org/10.1002/gepi.22205</a></li>
        <li>Wu, Y., et al. (2019). Genome-wide association study of gastrointestinal disorders reinforces the link between the digestive tract and the nervous system. BioRxiv, 811737. <a target="_blank" href="https://doi.org/10.1101/811737">https://doi.org/10.1101/811737</a></li>
        <li>Yang, F., et al. (2019). CCmed: cross-condition mediation analysis for identifying robust trans-eQTLs and assessing their effects on human traits. BioRxiv, 803106. <a target="_blank" href="https://doi.org/10.1101/803106">https://doi.org/10.1101/803106</a></li>
        <li>Yang, Y., et al. (2019). Genetic and Expression Analysis of COPI Genes and Alzheimer’s Disease Susceptibility. Frontiers in Genetics, 10. <a target="_blank" href="https://doi.org/10.3389/fgene.2019.00866">https://doi.org/10.3389/fgene.2019.00866</a></li>
        <li>Yu, H., et al. (2020). Integration analysis of methylation quantitative trait loci and GWAS identify three schizophrenia risk variants. Neuropsychopharmacology. <a target="_blank" href="https://doi.org/10.1038/s41386-020-0605-3">https://doi.org/10.1038/s41386-020-0605-3</a></li>
        <li>Zeng, B., et al. (2019). Comprehensive Multiple eQTL Detection and Its Application to GWAS Interpretation. Genetics. <a target="_blank" href="https://doi.org/10.1534/genetics.119.302091">https://doi.org/10.1534/genetics.119.302091</a></li>
        <li>Zheng, Z., et al. (2019). QTLbase: an integrative resource for quantitative trait loci across multiple human molecular phenotypes. Nucleic Acids Research. <a target="_blank" href="https://doi.org/10.1093/nar/gkz888">https://doi.org/10.1093/nar/gkz888</a></li>
        <li>Zhou, X., et al. (2019). Non-coding variability at the APOE locus contributes to the Alzheimer’s risk. Nature Communications, 10(1), 3310. <a target="_blank" href="https://doi.org/10.1038/s41467-019-10945-z">https://doi.org/10.1038/s41467-019-10945-z</a></li>
        <li>Zhu, X., et al. (2020). Modeling regulatory network topology improves genome-wide analyses of complex human traits. BioRxiv, 2020.03.13.990010. <a target="_blank" href="https://doi.org/10.1101/2020.03.13.990010">https://doi.org/10.1101/2020.03.13.990010</a></li>
      </ol>

      <h5>Papers citing eQTLGen</h5>
      <ol>
        <li value="84">Cai, M., et al. (2019). Quantifying the impact of genetically regulated expression on complex traits and diseases. BioRxiv, 546580. <a target="_blank" href="http://doi.org/10.1101/546580">http://doi.org/10.1101/546580</a></li>
        <li>Cheng, F.-F., et al., (2019). Towards the identification of causal genes for age-related macular degeneration. BioRxiv. <a target="_blank" href="https://doi.org/10.1101/778613">https://doi.org/10.1101/778613</a></li>
        <li>Hawe, J. S., et al. (2019). Inferring interaction networks from multi-omics data. Frontiers in Genetics, 10(JUN). <a target="_blank" href="https://doi.org/10.3389/fgene.2019.00535">https://doi.org/10.3389/fgene.2019.00535</a></li>
        <li>Kalayci, S., et al. (2019). ImmuneRegulation: a web-based tool for identifying human immune regulatory elements. Nucleic Acids Research, 47(W1), W142–W150. <a target="_blank" href="https://doi.org/10.1093/nar/gkz450">https://doi.org/10.1093/nar/gkz450</a></li>
        <li>Kerimov, N., et al. (2020). eQTL Catalogue: a compendium of uniformly processed human gene expression and splicing QTLs. BioRxiv, 2020.01.29.924266. <a target="_blank" href="https://doi.org/10.1101/2020.01.29.924266">https://doi.org/10.1101/2020.01.29.924266</a></li>
        <li>Lappalainen, T., et al. (2019). Genomic Analysis in the Age of Human Genome Sequencing. Cell, 177(1), 70–84. <a target="_blank" href="http://doi.org/10.1016/j.cell.2019.02.032">http://doi.org/10.1016/j.cell.2019.02.032</a></li>
        <li>Liu, X., et al. (2019). Trans Effects on Gene Expression Can Drive Omnigenic Inheritance. Cell, 177(4), 1022-1034.e6. <a target="_blank" href="https://doi.org/10.1016/j.cell.2019.04.014">https://doi.org/10.1016/j.cell.2019.04.014</a></li>
        <li>Lu, L., et al. (2019). Robust Hi-C chromatin loop maps in human neurogenesis and brain tissues at high-resolution. BioRxiv, 744540. <a target="_blank" href="https://doi.org/10.1101/744540">https://doi.org/10.1101/744540</a></li>
        <li>Neumeyer, S., et al (2019). Strengthening Causal Inference for Complex Disease Using Molecular Quantitative Trait Loci. Trends in Molecular Medicine, 1–10. <a target="_blank" href="https://doi.org/10.1016/j.molmed.2019.10.004">https://doi.org/10.1016/j.molmed.2019.10.004</a></li>
        <li>Rotival, M. (2019). Characterising the genetic basis of immune response variation to identify causal mechanisms underlying disease susceptibility. HLA, 94(3), 275–284. <a target="_blank" href="https://doi.org/10.1111/tan.13598">https://doi.org/10.1111/tan.13598</a></li>
        <li>Sidorenko, J., et al. (2019). The effect of X-linked dosage compensation on complex trait variation. Nature Communications, 10(1), 3009. <a target="_blank" href="https://doi.org/10.1038/s41467-019-10598-y">https://doi.org/10.1038/s41467-019-10598-y</a></li>
        <li>Storm, C. S. et al. (2020). Using Mendelian randomization to understand and develop treatments for neurodegenerative disease. Brain Communications. <a target="_blank" href="https://doi.org/10.1093/BRAINCOMMS/FCAA031">https://doi.org/10.1093/BRAINCOMMS/FCAA031</a></li>
        <li>Wainberg, M., et al. (2019). Opportunities and challenges for transcriptome-wide association studies. Nature Genetics 2019 51:4, 51(4), 592. <a target="_blank" href="http://doi.org/10.1038/s41588-019-0385-z">http://doi.org/10.1038/s41588-019-0385-z</a></li>
        <li>van der Wijst, M. G., et al. (2020). The single-cell eQTLGen consortium. ELife, 9. <a target="_blank" href="https://doi.org/10.7554/eLife.52155">https://doi.org/10.7554/eLife.52155</a></li>
        <li>Yao, D. W., et al. (2019). Quantifying genetic effects on disease mediated by assayed gene expression levels. BioRxiv, 730549. <a target="_blank" href="https://doi.org/10.1101/730549">https://doi.org/10.1101/730549</a></li>
        <li>Ye, Y., et al. (2020). A Multi-Omics Perspective of Quantitative Trait Loci in Precision Medicine. Trends in Genetics. <a target="_blank" href="https://doi.org/10.1016/j.tig.2020.01.009">https://doi.org/10.1016/j.tig.2020.01.009</a></li>
        <li>Yeung, K.-F., et al. (2019). CoMM: A Collaborative Mixed Model That Integrates GWAS and eQTL Data Sets to Investigate the Genetic Architecture of Complex Traits. Bioinformatics and Biology Insights, 13, 117793221988143. <a target="_blank" href="https://doi.org/10.1177/1177932219881435">https://doi.org/10.1177/1177932219881435</a></li>
        <li>Zheng, J., et al (2019, November 21). Use of Mendelian Randomization to Examine Causal Inference in Osteoporosis. Frontiers in Endocrinology. Frontiers Media S.A. <a target="_blank" href="https://doi.org/10.3389/fendo.2019.00807">https://doi.org/10.3389/fendo.2019.00807</a></li>
        <li>Zwir, I., et al. (2019). Uncovering the complex genetics of human personality: response from authors on the PGMRA Model. Molecular Psychiatry, 1. <a target="_blank" href="http://doi.org/10.1038/s41380-019-0399-z">http://doi.org/10.1038/s41380-019-0399-z</a></li>
      </ol>

      <h5>Computational methods</h5>
      Online computational methods that use eQTLGen data and came available after submitting our initial manuscript:<br/>
      <ul>
        <li><a target="_blank" href="http://fuma.ctglab.nl/">http://fuma.ctglab.nl/</a></li>
        <li><a target="_blank" href="http://locuscompare.com/">http://locuscompare.com/</a></li>
        <li><a target="_blank" href="https://kalays01.u.hpc.mssm.edu/hipc/">https://kalays01.u.hpc.mssm.edu/hipc/</a></li>
        <li><a target="_blank" href="http://mrcieu.mrsoftware.org/Tissue_MR_atlas/">http://mrcieu.mrsoftware.org/Tissue_MR_atlas/</a></li>
        <li><a target="_blank" href="https://immuneregulation.mssm.edu">https://immuneregulation.mssm.edu</a></li>
        <li><a target="_blank" href="http://mulinlab.tmu.edu.cn/qtlbase">http://mulinlab.tmu.edu.cn/qtlbase</a></li>
        <li><a target="_blank" href="http://www.epigraphdb.org/xqtl/">http://www.epigraphdb.org/xqtl/</a></li>
        <li><a target="_blank" href="http://www.phenoscanner.medschl.cam.ac.uk">http://www.phenoscanner.medschl.cam.ac.uk</a></li>
      </ul>

    </div>
    <!-- /.container -->

    <!-- Footer -->
    <footer class="pt-5">
      <hr>
      <div class="container text-center">
        <img src="img/rug.jpg" height="100px">
        <p class="m-0 text-center text-white">Copyright &copy; eQTLGen 2018</p>
      </div>
      <!-- /.container -->
    </footer>

    <!-- Bootstrap core JavaScript -->
    <script src="js/vendor/jquery/jquery.min.js"></script>
    <script src="js/vendor/bootstrap/js/bootstrap.bundle.min.js"></script>

  </body>

</html>
